File: C++/Jump-Game-II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int sc = 0, end = 0, maxsteps = 0;
        for (int i = 0; i < nums.size() - 1; i ++) {
            maxsteps = max(maxsteps, nums[i] + i);
            if (i == end) {
                end = maxsteps;
                sc++;
            }
        }
        return sc;
    }
};

File: Python/Jump-Game-II.py
class Solution:
    # @param A, a list of integers
    # @return an integer
    def jump(self, A):
        ret = 0
        last = 0
        curr = 0
        for i in range(len(A)):
            if i > last:
                last = curr
                ret += 1
            curr = max(curr, i+A[i])
        return ret